code/workspaces/client-api/src/auth/permissionChecker.spec.js

```diff
@@ -14,60 +16,119 @@ const admin = {
       .then(() => {
         expect(actionMock).toHaveBeenCalledTimes(1);
         expect(actionMock).toHaveBeenCalledWith('value');
       }));
   });
 
   describe('multiPermissionWrapper', () => {
-    it('throws an error if user is lacking correct permission', () => multiPermissionsWrapper(['elementName:missingActionName', 'elementName:anotherAction'], user, () => actionMock('value'))
+    it('throws an error if user is lacking correct permission', () => multiPermissionsWrapper(['elementName:missingActionName', 'elementName:anotherAction'], user, next)
       .catch((err) => {
         expect(err).toEqual(new Error('User missing expected permission(s): project:elementName:missingActionName,project:elementName:anotherAction,system:instance:admin'));
         expect(actionMock).not.toHaveBeenCalled();
       }));
 
-    it('callback to be called if user has correct permission', () => multiPermissionsWrapper(['elementName:actionName', 'elementName:anotherAction'], user, () => actionMock('value'))
+    it('callback to be called if user has correct permission', () => multiPermissionsWrapper(['elementName:actionName', 'elementName:anotherAction'], user, next)
       .then(() => {
         expect(actionMock).toHaveBeenCalledTimes(1);
         expect(actionMock).toHaveBeenCalledWith('value');
       }));
 
-    it('callback to be called if user has instance admin permission', () => multiPermissionsWrapper(['elementName:actionName'], admin, () => actionMock('value'))
+    it('callback to be called if user has instance admin permission', () => multiPermissionsWrapper(['elementName:actionName'], admin, next)
       .then(() => {
         expect(actionMock).toHaveBeenCalledTimes(1);
         expect(actionMock).toHaveBeenCalledWith('value');
       }));
   });
 
   describe('instanceAdminWrapper', () => {
-    it('throws an error if user is lacking correct permission', () => instanceAdminWrapper(user, () => actionMock('value'))
+    it('throws an error if user is lacking correct permission', () => instanceAdminWrapper(user, next)
       .catch((err) => {
         expect(err).toEqual(new Error('User missing expected permission(s): system:instance:admin'));
         expect(actionMock).not.toHaveBeenCalled();
       }));
 
-    it('callback to be called if user has instance admin permission', () => instanceAdminWrapper(admin, () => actionMock('value'))
+    it('callback to be called if user has instance admin permission', () => instanceAdminWrapper(admin, next)
       .then(() => {
         expect(actionMock).toHaveBeenCalledTimes(1);
         expect(actionMock).toHaveBeenCalledWith('value');
       }));
   });
+
+  describe('projectPermissionWrapper', () => {
+    const httpMock = new MockAdapter(axios);
+
+    beforeEach(() => {
+      httpMock.reset();
+    });
+
+    afterAll(() => {
+      httpMock.restore();
+    });
+
+    it('throws an error if user is lacking user permission', () => {
```

Comment:
`async/await` might also make this a bit clearer.
```js
it('throws an error if user is lacking user permission', async () => {
  try {
    await projectPermissionWrapper('project2', 'elementName:missingActionName', 'token', user, next);
  } catch (err) {
    expect(err).toEqual(new Error('Requested projectKey projectX not accessible to user'));
	  expect(actionMock).not.toHaveBeenCalled();
  }
});
```

Would this test still pass if no error is thrown? If there is no error then it won't enter the catch so there will be no `expect`s called. Don't know whether you would need to have a `let error` and then assign it in the catch like
```js
it('throws an error if user is lacking user permission', async () => {
  let error;
  try {
    await projectPermissionWrapper('project2', 'elementName:missingActionName', 'token', user, next);
  } catch (err) {
    error = err;
  }
  expect(error).toEqual(new Error('Requested projectKey projectX not accessible to user'));
  expect(actionMock).not.toHaveBeenCalled();
});
```
I think the same idea applies to the other tests that have been added. For the ones that check the `.then` I don't think they would be called if an error was thrown (although I would except the test to fail as the error will not have been caught so I guess it would still work as intended).

Reply:
Tests updated as suggested, and approach used in other tests in that spec.